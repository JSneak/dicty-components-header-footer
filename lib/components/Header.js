"use strict";

exports.__esModule = true;

var _react = require("react");

var _react2 = _interopRequireDefault(_react);

var _logo = require("../images/logo.png");

var _logo2 = _interopRequireDefault(_logo);

var _Link = require("../styles/Link");

var _Link2 = _interopRequireDefault(_Link);

var _headerStyles = require("../styles/headerStyles");

var _styles = require("@material-ui/core/styles");

var _Grid = require("@material-ui/core/Grid");

var _Grid2 = _interopRequireDefault(_Grid);

var _ExpandedSearch = require("./ExpandedSearch");

var _ExpandedSearch2 = _interopRequireDefault(_ExpandedSearch);

var _NormalSearch = require("./NormalSearch");

var _NormalSearch2 = _interopRequireDefault(_NormalSearch);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

// set base URL for homepage
var Home = "/";

/**
 * The `Header` component that will be displayed
 * on top of the navigation bar in every React web application of [dictyBase](http://dictybase.org).
 */

var Header = function Header(props) {
  var _useState = (0, _react.useState)(false),
      isExpanded = _useState[0],
      setIsExpanded = _useState[1];

  var classes = props.classes,
      _props$home = props.home,
      home = _props$home === undefined ? Home : _props$home,
      children = props.children,
      items = props.items;


  var handleClick = function handleClick() {
    setIsExpanded(!isExpanded);
  };

  return _react2.default.createElement(
    _Grid2.default,
    { container: true, className: classes.root },
    _react2.default.createElement(
      _Grid2.default,
      { item: true, xs: 12, sm: 5, md: 3, lg: 3, className: classes.logoContainer },
      _react2.default.createElement(
        _Link2.default,
        { href: home },
        _react2.default.createElement("img", { src: _logo2.default, alt: "dictyBase logo", className: classes.image })
      )
    ),
    isExpanded ? _react2.default.createElement(_ExpandedSearch2.default, { isExpanded: isExpanded }) : _react2.default.createElement(_NormalSearch2.default, { handleClick: handleClick }),
    _react2.default.createElement(
      _Grid2.default,
      {
        item: true,
        xs: 12,
        sm: 12,
        md: 4,
        lg: 4,
        className: classes.linkContainer },
      children(items)
    )
  );
};

exports.default = (0, _styles.withStyles)(_headerStyles.headerStyles)(Header);
module.exports = exports["default"];