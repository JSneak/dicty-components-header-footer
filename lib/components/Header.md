## Default header component

```example
        <Header
          downloads="/downloads"
          info="/information"
          cite="/citation"
        />
```
