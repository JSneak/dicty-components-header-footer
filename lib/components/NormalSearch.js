"use strict";

exports.__esModule = true;

var _react = require("react");

var _react2 = _interopRequireDefault(_react);

var _headerStyles = require("../styles/headerStyles");

var _styles = require("@material-ui/core/styles");

var _Grid = require("@material-ui/core/Grid");

var _Grid2 = _interopRequireDefault(_Grid);

var _Input = require("@material-ui/core/Input");

var _Input2 = _interopRequireDefault(_Input);

var _InputLabel = require("@material-ui/core/InputLabel");

var _InputLabel2 = _interopRequireDefault(_InputLabel);

var _InputAdornment = require("@material-ui/core/InputAdornment");

var _InputAdornment2 = _interopRequireDefault(_InputAdornment);

var _FormControl = require("@material-ui/core/FormControl");

var _FormControl2 = _interopRequireDefault(_FormControl);

var _Search = require("@material-ui/icons/Search");

var _Search2 = _interopRequireDefault(_Search);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/**
 * NormalSearch handles the unexpanded appearance of the search box.
 */

var NormalSearch = function NormalSearch(props) {
  var classes = props.classes,
      handleClick = props.handleClick;


  return _react2.default.createElement(
    _Grid2.default,
    { item: true, xs: 12, sm: 7, md: 4, lg: 4, className: classes.searchContainer },
    _react2.default.createElement(
      _FormControl2.default,
      { className: classes.textField },
      _react2.default.createElement(
        _InputLabel2.default,
        { htmlFor: "guided-search" },
        "Guided Search (coming soon)"
      ),
      _react2.default.createElement(_Input2.default, {
        id: "input-with-icon-adornment",
        onClick: handleClick,
        endAdornment: _react2.default.createElement(
          _InputAdornment2.default,
          { className: classes.searchIcon, position: "end" },
          _react2.default.createElement(_Search2.default, null)
        )
      })
    )
  );
};
exports.default = (0, _styles.withStyles)(_headerStyles.headerStyles)(NormalSearch);
module.exports = exports["default"];