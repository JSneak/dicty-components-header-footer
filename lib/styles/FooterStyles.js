"use strict";

exports.__esModule = true;
var FooterStyles = exports.FooterStyles = function FooterStyles(theme) {
  var _ref;

  return _ref = {
    root: {
      width: "100%",
      fontFamily: "Roboto, sans-serif",
      background: "linear-gradient(#2c3e50, #3498db)",
      paddingLeft: 20,
      paddingRight: 20,
      paddingBottom: 5
    },
    listItem: {
      margin: 0,
      padding: 1,
      listStyle: "none"
    },
    link: {
      color: "#d8d8d8",
      fontSize: 11,
      textDecoration: "none",
      "&:hover": {
        textDecoration: "underline"
      }
    },
    subsectionItem: {
      paddingLeft: 10,
      paddingRight: 10
    },
    ulHeader: {
      color: "#19d6a0",
      fontSize: 14,
      textDecoration: "none"
    }
  }, _ref[theme.breakpoints.down("sm")] = {
    listItem: {
      textAlign: "center"
    },
    headerLink: {
      textAlign: "center"
    }
  }, _ref[theme.breakpoints.down("xs")] = {
    root: {
      padding: 0
    },
    gridItem: {
      width: "50%"
    },
    subsectionItem: {
      paddingLeft: 0
    }
  }, _ref;
};