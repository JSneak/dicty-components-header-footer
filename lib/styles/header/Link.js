"use strict";

exports.__esModule = true;

var _templateObject = _taggedTemplateLiteralLoose(["\n  color: #15317e;\n  margin-right: 8px;\n  text-decoration: none;\n"], ["\n  color: #15317e;\n  margin-right: 8px;\n  text-decoration: none;\n"]);

var _react = require("react");

var _react2 = _interopRequireDefault(_react);

var _styledComponents = require("styled-components");

var _styledComponents2 = _interopRequireDefault(_styledComponents);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

function _taggedTemplateLiteralLoose(strings, raw) { strings.raw = raw; return strings; }

var LinkStyle = _styledComponents2.default.a(_templateObject);
var Link = function Link(_ref) {
  var children = _ref.children,
      props = _objectWithoutProperties(_ref, ["children"]);

  return _react2.default.createElement(
    LinkStyle,
    props,
    children
  );
};

exports.default = Link;
module.exports = exports["default"];