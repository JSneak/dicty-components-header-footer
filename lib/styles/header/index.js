"use strict";

exports.__esModule = true;

var _Img = require("./Img");

Object.defineProperty(exports, "Img", {
  enumerable: true,
  get: function get() {
    return _interopRequireDefault(_Img).default;
  }
});

var _Link = require("./Link");

Object.defineProperty(exports, "Link", {
  enumerable: true,
  get: function get() {
    return _interopRequireDefault(_Link).default;
  }
});

var _LinksContainer = require("./LinksContainer");

Object.defineProperty(exports, "LinksContainer", {
  enumerable: true,
  get: function get() {
    return _interopRequireDefault(_LinksContainer).default;
  }
});

var _Logo = require("./Logo");

Object.defineProperty(exports, "Logo", {
  enumerable: true,
  get: function get() {
    return _interopRequireDefault(_Logo).default;
  }
});

var _Magnifier = require("./Magnifier");

Object.defineProperty(exports, "Magnifier", {
  enumerable: true,
  get: function get() {
    return _interopRequireDefault(_Magnifier).default;
  }
});

var _Right = require("./Right");

Object.defineProperty(exports, "Right", {
  enumerable: true,
  get: function get() {
    return _interopRequireDefault(_Right).default;
  }
});

var _SearchBar = require("./SearchBar");

Object.defineProperty(exports, "SearchBar", {
  enumerable: true,
  get: function get() {
    return _interopRequireDefault(_SearchBar).default;
  }
});

var _SearchContainer = require("./SearchContainer");

Object.defineProperty(exports, "SearchContainer", {
  enumerable: true,
  get: function get() {
    return _interopRequireDefault(_SearchContainer).default;
  }
});

var _Container = require("./Container");

Object.defineProperty(exports, "Container", {
  enumerable: true,
  get: function get() {
    return _interopRequireDefault(_Container).default;
  }
});

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }