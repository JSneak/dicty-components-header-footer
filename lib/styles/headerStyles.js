"use strict";

exports.__esModule = true;
var headerStyles = exports.headerStyles = function headerStyles(theme) {
  var _ref;

  return _ref = {
    root: {
      padding: 5,
      fontFamily: "Roboto, sans-serif"
    },
    logoContainer: {
      paddingBottom: 0
    },
    dcr: {
      fontWeight: 400,
      fontSize: "1.5rem",
      // color: "#004080", #142a70
      color: "#54575c",
      paddingLeft: "18px"
    },
    image: {
      maxWidth: 220,
      height: 78
    },
    searchContainer: {
      display: "flex",
      justifyContent: "flex-end",
      height: "40px"
    },
    textField: {
      width: "50%",
      paddingBottom: 0
    },
    textFieldExpanded: {
      width: "75%",
      paddingBottom: 0
    },
    linkContainer: {
      display: "flex",
      justifyContent: "flex-end",
      paddingBottom: 0,
      fontSize: "0.8em",
      height: "25px"
    },
    searchLabel: {
      "&$searchFocused": {
        color: "#004080"
      }
    },
    searchFocused: {},
    searchUnderline: {
      "&:after": {
        borderBottomColor: "#004080"
      }
    },
    searchIcon: {
      color: "#004080"
    }
  }, _ref[theme.breakpoints.down("sm")] = {
    textField: {
      marginTop: 0,
      width: "85%"
    },
    textFieldExpanded: {
      marginTop: 0
    },
    logoContainer: {
      display: "flex",
      justifyContent: "center"
    },
    linkContainer: {
      justifyContent: "center"
    }
  }, _ref[theme.breakpoints.only("sm")] = {
    searchContainer: {
      marginTop: 12
    }
  }, _ref;
};