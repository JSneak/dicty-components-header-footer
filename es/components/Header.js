function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

import React, { Component } from "react";
import logo from "../images/logo.png";
import Link from "../styles/Link";
import { headerStyles as styles } from "../styles/headerStyles";
import { withStyles } from "@material-ui/core/styles";
import Grid from "@material-ui/core/Grid";
import Input from "@material-ui/core/Input";
import InputLabel from "@material-ui/core/InputLabel";
import InputAdornment from "@material-ui/core/InputAdornment";
import FormControl from "@material-ui/core/FormControl";
import Grow from "@material-ui/core/Grow";
import Search from "@material-ui/icons/Search";

// set base URL for homepage
var Home = "http://dictybase.org";

/**
 * The `Header` component that will be displayed
 * on top of navigation bar in every React web application of [dictyBase](http://dictybase.org).
 */

var Header = function (_Component) {
  _inherits(Header, _Component);

  function Header() {
    var _temp, _this, _ret;

    _classCallCheck(this, Header);

    for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    return _ret = (_temp = (_this = _possibleConstructorReturn(this, _Component.call.apply(_Component, [this].concat(args))), _this), _this.state = {
      isExpanded: false
    }, _this.onClick = function () {
      _this.setState({ isExpanded: !_this.state.isExpanded });
    }, _temp), _possibleConstructorReturn(_this, _ret);
  }

  Header.prototype.render = function render() {
    var _props = this.props,
        classes = _props.classes,
        _props$home = _props.home,
        home = _props$home === undefined ? Home : _props$home,
        children = _props.children,
        items = _props.items;
    var isExpanded = this.state.isExpanded;

    return React.createElement(
      Grid,
      { container: true, spacing: 8, className: classes.root },
      React.createElement(
        Grid,
        {
          item: true,
          xs: 12,
          sm: 5,
          md: 3,
          lg: 3,
          className: classes.logoContainer },
        React.createElement(
          Link,
          { href: home },
          React.createElement("img", { src: logo, alt: "dictyBase logo", className: classes.image })
        )
      ),
      !isExpanded && React.createElement(
        Grid,
        {
          item: true,
          xs: 12,
          sm: 7,
          md: 5,
          lg: 5,
          className: classes.searchContainer },
        React.createElement(
          FormControl,
          { className: classes.textField },
          React.createElement(
            InputLabel,
            { htmlFor: "guided-search" },
            "Guided Search"
          ),
          React.createElement(Input, {
            id: "input-with-icon-adornment",
            onClick: this.onClick,
            endAdornment: React.createElement(
              InputAdornment,
              { position: "end" },
              React.createElement(Search, null)
            )
          })
        )
      ),
      isExpanded && React.createElement(
        Grow,
        { "in": isExpanded, timeout: 500, style: { opacity: 1 } },
        React.createElement(
          Grid,
          {
            item: true,
            xs: 12,
            sm: 7,
            md: 5,
            lg: 5,
            className: classes.searchContainer },
          React.createElement(
            FormControl,
            { className: classes.textFieldExpanded },
            React.createElement(
              InputLabel,
              { htmlFor: "guided-search" },
              "Guided Search"
            ),
            React.createElement(Input, {
              id: "input-with-icon-adornment",
              autoFocus: true,
              endAdornment: React.createElement(
                InputAdornment,
                { position: "end" },
                React.createElement(Search, null)
              )
            })
          )
        )
      ),
      React.createElement(
        Grid,
        {
          item: true,
          xs: 12,
          sm: 12,
          md: 4,
          lg: 4,
          className: classes.linkContainer },
        children(items)
      )
    );
  };

  return Header;
}(Component);

export default withStyles(styles)(Header);