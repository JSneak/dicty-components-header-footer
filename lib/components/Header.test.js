import React from "react"
import { shallow, configure } from "enzyme"
import Adapter from "enzyme-adapter-react-16"
import Header from "./Header"
import Link from "../styles/Link"
import { styled } from "@material-ui/styles"
import items from "../data/header"
import "font-awesome/css/font-awesome.min.css"
import FontAwesome from "react-fontawesome"
import Grid from "@material-ui/core/Grid"
import NormalSearch from "./NormalSearch"

configure({ adapter: new Adapter() })

const RouterLink = styled(Link)({
  color: "#ff6b81",
})

const generateLinks = (link, i) =>
  link.isRouter ? (
    <RouterLink key={i} href={link.url}>
      <FontAwesome name={link.icon} />
      {link.text}
    </RouterLink>
  ) : (
    <Link key={i} href={link.url}>
      <FontAwesome name={link.icon} />
      {link.text}
    </Link>
  )

describe("Header", () => {
  const wrapper = shallow(
    <Header items={items}>{items => items.map(generateLinks)}</Header>,
  ).dive()

  describe("initial render", () => {
    it("renders without crashing", () => {
      expect(wrapper).toHaveLength(1)
    })

    it("renders initial components", () => {
      expect(wrapper.find(Grid)).toHaveLength(3)
      expect(wrapper.find(NormalSearch)).toHaveLength(1)
    })
  })
})
