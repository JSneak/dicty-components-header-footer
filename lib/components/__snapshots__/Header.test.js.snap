// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matching a snapshot of Header 1`] = `
<div
  className="MuiGrid-root Header-root-1 MuiGrid-container MuiGrid-spacing-xs-8"
>
  <div
    className="MuiGrid-root Header-logoContainer-2 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-5 MuiGrid-grid-md-3 MuiGrid-grid-lg-3"
  >
    <a
      className="a-root-115"
      href="/"
    >
      <img
        alt="dictyBase logo"
        className="Header-image-3"
        src="test-file-stub"
      />
    </a>
  </div>
  <div
    className="MuiGrid-root Header-searchContainer-4 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-7 MuiGrid-grid-md-5 MuiGrid-grid-lg-5"
  >
    <div
      className="MuiFormControl-root Header-textField-5"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
        data-shrink={false}
        htmlFor="guided-search"
      >
        Guided Search (coming soon)
      </label>
      <div
        className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedEnd"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          className="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd"
          disabled={false}
          id="input-with-icon-adornment"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={false}
          type="text"
        />
        <div
          className="MuiInputAdornment-root Header-searchIcon-11 MuiInputAdornment-positionEnd"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
            />
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
  <div
    className="MuiGrid-root Header-linkContainer-7 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-4 MuiGrid-grid-lg-4"
  >
    <a
      className="a-root-115 Link-root-181"
      href="/login"
    >
      <span
        aria-hidden={true}
        className="fa fa-sign-in"
      />
      Login
    </a>
    <a
      className="a-root-115"
      href="/cite"
    >
      <span
        aria-hidden={true}
        className="fa fa-plus"
      />
      Cite Us
    </a>
    <a
      className="a-root-115"
      href="/downloads"
    >
      <span
        aria-hidden={true}
        className="fa fa-download"
      />
      Downloads
    </a>
    <a
      className="a-root-115"
      href="/info"
    >
      <span
        aria-hidden={true}
        className="fa fa-info-circle"
      />
      About dictyBase
    </a>
  </div>
</div>
`;
